src/aap_eda/api/serializers/activation.py

```diff
@@ -383,31 +402,28 @@ def create(self, validated_data):
                 validated_data
             )
 
+        vault_data = VaultData()
+
         if validated_data.get("event_streams"):
             validated_data["rulebook_rulesets"] = _updated_ruleset(
-                validated_data
-            )
-
-        extra_var_obj = None
-        password = secrets.token_urlsafe()
-        system_vault_credential_needed = False
-
-        if validated_data.get("extra_var_id"):
-            extra_var_obj = models.ExtraVar.objects.get(
-                id=validated_data["extra_var_id"]
+                validated_data, vault_data
             )
 
         vault = _get_vault_credential_type()
 
         if validated_data.get("eda_credentials"):
-            system_vault_credential_needed = _handle_eda_credentials(
-                validated_data, extra_var_obj, password
+            _update_extra_vars_from_eda_credentials(
+                validated_data, vault_data, True
```

Comment:
Can you use named arguments as you do in other places? passing booleans as positional arguments is misleading and considered a bad practice.